### PR TITLE
Amplify Modularization

### PR DESCRIPTION
**Pull request by jordanranz**
*Description of changes:*
- Modularization work from @powerful23 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/storage/__tests__/Providers/AWSS3Provider-unit-test.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@
  * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
  * and limitations under the License.
  */
-import  StorageProvider  from '../../src/Providers/AWSS3Provider';
+import StorageProvider from '../../src/providers/AWSS3Provider';
```
Comment (bluezebragames) [inline packages/storage/__tests__/Providers/AWSS3Provider-unit-test.ts]: With all of these `Providers`->`providers` changes, maybe change the name of the directory these tests are in too?

File: packages/storage/build.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+'use strict'
+
+const type = process.argv[2];
+
+const build = require('../../scripts/build');
+
+build(process.argv[2]);
```
Comment (bluezebragames) [inline packages/storage/build.js]:
```suggestion
build(type);
```

File: packages/xr/build.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+'use strict'
+
+const type = process.argv[2];
+
+const build = require('../../scripts/build');
+
+build(process.argv[2]);
```
Comment (bluezebragames) [inline packages/xr/build.js]:
```suggestion
build(type);
```

File: packages/xr/src/XR.ts [hunk 4]
```diff
@@ -19,7 +19,7 @@ const logger = new Logger('XR');
 
 const DEFAULT_PROVIDER_NAME = "SumerianProvider";
 
-export default class XR {
+export class XRClass {
```
Comment (bluezebragames) [inline packages/xr/src/XR.ts]: See above

File: packages/xr/src/index.ts [hunk 5]
```diff
@@ -23,13 +23,16 @@ if (!_instance) {
...
 }
 
-const XR = _instance;
+export const XR = _instance;
 Amplify.register(XR);
 
+/**
+ * @deprecated use named import
+ */
 export default XR;
```
Comment (bluezebragames) [inline packages/xr/src/index.ts]: Export default last?

File: packages/analytics/__tests__/Providers/AWSAnalyticsProvider-unit-test.ts [hunk 6]
```diff
@@ -0,0 +1,830 @@
...
+                        },
+                        "UserId":"default"
+                    }
+                }
+            });
+
+            spyon.mockClear();
+        });
+
+        test('happy case with specified enpoint configure provided', async () => {
```
[the remark this answers is not in the document]
Comment (jordanranz) [reply packages/analytics/__tests__/Providers/AWSAnalyticsProvider-unit-test.ts]: good catch

**Comment by jadbox** [comment]
Does this relate to (or is a duplicate of) https://github.com/aws-amplify/amplify-js/issues/122 ?

**Comment by jadbox** [comment]
Is this in addition to "Modularization work from @powerful23"?

**Comment by jadbox** [comment]
It's been a few months- any update on this?

**Comment by sammartinez** [comment]
Closing this as I don't believe this pull request is needed anymore

**Pull request closed without merging**